src/packages/admin-ui-components/src/detail-panel/fields/media-field/media-field.tsx

```diff
@@ -0,0 +1,80 @@
+import { useField, useFormikContext } from 'formik';
+import { EntityField } from '../../../utils';
+import { useMutation } from '@apollo/client';
+import { getUploadUrlMutation } from '../../graphql';
+import { useState } from 'react';
+import styles from './styles.module.css';
+import { Button } from '../../../button';
+
+export const MediaField = ({ field }: { field: EntityField }) => {
+	const { setValues } = useFormikContext();
+	const [mediaHasChanged, setMediaHasChanged] = useState(false);
+	const [_, meta] = useField({ name: field.name, multiple: false });
+	const { initialValue: downloadUrl } = meta;
+	const [getUploadUrl] = useMutation(getUploadUrlMutation);
+
+	const handleFileUpload = async (file: any) => {
+		const res = await getUploadUrl({ variables: { key: file.name } });
+		const uploadURL = res.data.getUploadUrl;
+		if (!uploadURL) {
+			console.error('Upload URL is not available');
+			return;
+		}
+
+		if (!field.extensions?.key) {
+			console.error('Key not found on field extentions');
+			return;
+		}
+		const mediaKey = field.extensions.key;
+
+		setValues((prev: any) => ({
+			...prev,
+			uploadUrl: uploadURL,
+			file: file,
+			// overwrite the key form field value with the field.extensions.key
+			[mediaKey]: file.name,
+		}));
+		setMediaHasChanged(true);
+	};
+
+	const handleFileInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		if (event.target.files) {
+			handleFileUpload(event.target.files[0]);
+		}
+	};
+
+	const handleOnDelete = () => {
+		setValues((prev: any) => ({
+			...prev,
+			key: null,
+			uploadUrl: null,
+			file: null,
+			downloadUrl: null,
```

Comment:
Im not sure if `downloadUrl` refers to this https://github.com/exogee-technology/graphweaver/pull/274/files#diff-1c99eda95dafecacb997936b67bff1adacfe681874acc6e916aee0b64e03c7f3R33

Could you test changing the attribute name on the backend to confirm it works?

Reply:
No these are diff, will change, see above.